frontend/src/test/scala/bloop/dap/DebugProtocolSpec.scala

```diff
@@ -0,0 +1,142 @@
+package bloop.dap
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+
+import bloop.bsp.BspBaseSuite
+import bloop.cli.BspProtocol
+import bloop.logging.RecordingLogger
+import bloop.util.{TestProject, TestUtil}
+
+object DebugProtocolSpec extends BspBaseSuite {
+  override val protocol: BspProtocol = BspProtocol.Local
+
+  test("starts a debug session") {
+    TestUtil.withinWorkspace { workspace =>
+      val main =
+        """|/main/scala/Main.scala
+           |object Main {
+           |  def main(args: Array[String]): Unit = {
+           |    println("Hello, World!")
+           |  }
+           |}
+           |""".stripMargin
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val project = TestProject(workspace, "p", List(main))
+
+      loadBspState(workspace, List(project), logger) { state =>
+        val output = state.withDebugSession(project, "Main") { client =>
+          for {
+            _ <- client.initialize()
+            _ <- client.launch()
+            _ <- client.configurationDone()
+            _ <- client.exited
+            _ <- client.terminated
+            _ <- client.disconnect()
+            output <- client.allOutput
+          } yield output
+        }
+
+        assertNoDiff(output, "Hello, World!\n")
+      }
+    }
+  }
+
+  // when the session detaches from the JVM, the JDI once again writes to the standard output
+  test("restarted session does not contain JDI output") {
```

Comment:
This is the test case that I've mentioned before in the other comment and that currently fails.